impacted-tests-executor/src/main/java/org/junit/platform/console/tasks/TestIdentifierUtils.java

```diff
@@ -0,0 +1,42 @@
+package org.junit.platform.console.tasks;
+
+import org.junit.platform.console.Logger;
+import org.junit.platform.engine.support.descriptor.MethodSource;
+import org.junit.platform.launcher.TestIdentifier;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/** Utility classes for dealing with JUnit {@link TestIdentifier}s. */
+public class TestIdentifierUtils {
+
+	/**
+	 * Pattern that matches the classes fully qualified class name in JUnit's uniqueId as first capture group.
+	 * The test's uniqueId is something similar to:
+	 * [engine:junit-jupiter]/[class:com.example.project.JUnit5Test]/[method:testAdd()]
+	 * [engine:junit-vintage]/[runner:com.example.project.JUnit4Test]/[test:testAdd(com.example.project.JUnit4Test)]
+	 */
+	private static final Pattern FULL_CLASS_NAME_PATTERN = Pattern.compile(".*\\[(?:class|runner):([^]]+)\\].*");
+
+	/** Builds the uniform path which will later be displayed in Teamscale. */
+	public static String getTestUniformPath(TestIdentifier testIdentifier, Logger logger) {
```

Comment:
Not sure whether this could be deleted at some point, but the legacy reporting name best matches what I personally would expect the name to be: "test(some, params)".
The only other possibility would be getDisplayName, which uses a similar format, but could also fallback to use the @DisplayName if given, which in turn allows emojis and other stuff and I'm not sure whether the display name must be unique. Using this as uniform path seems also dangerous to me.